src/scripts/main.js

```diff
@@ -1,3 +1,244 @@
+addRowSelect(employeesTable);
+addTableEditing(employeesTable);
+
+const employeeForm = createForm('new-employee-form');
+
+document.body.append(employeeForm);
+
+addRowAppending(employeeForm, employeesTable);
+
+function addTableSorting(table) {
+  table.tHead.addEventListener('click', (event) => {
+    sortTable(table, event.target);
+  });
+
+  function sortTable(sortedTable, sortedColumn) {
+    const columnIndex = sortedColumn.cellIndex;
+    const tBody = sortedTable.tBodies[0];
+
+    const sortedRows = [...tBody.rows].sort((firstRow, secondRow) => {
+      const firstRowText = firstRow.cells[columnIndex].textContent;
+      const secondRowText = secondRow.cells[columnIndex].textContent;
+
+      if (getNumber(firstRowText)) {
+        return sortedColumn.isSortOrderChanged
+          ? getNumber(secondRowText) - getNumber(firstRowText)
+          : getNumber(firstRowText) - getNumber(secondRowText);
+      }
+
+      return sortedColumn.isSortOrderChanged
+        ? secondRowText.localeCompare(firstRowText)
+        : firstRowText.localeCompare(secondRowText);
+    });
+
+    tBody.append(...sortedRows);
+
+    sortedColumn.isSortOrderChanged = !sortedColumn.isSortOrderChanged;
+  }
+
+  function getNumber(string) {
+    return +string.replace(/\D*/g, '');
+  };
+}
+
+function addRowSelect(table) {
+  table.tBodies[0].addEventListener('click', (event) => {
+    const chosenRow = event.target.closest('TR');
+
+    if (!chosenRow) {
+      return;
+    }
+
+    const previousChosenRow = document.querySelector('tr.active');
+
+    if (previousChosenRow) {
+      previousChosenRow.classList.remove('active');
+    }
+
+    chosenRow.classList.add('active');
+  });
+}
+
+function createForm(className) {
+  const form = document.createElement('form');
+
+  form.classList.add(className);
+
+  form.innerHTML = `
+    <label for="name">
+      Name:
+      <input
+        name="name"
+        id="name"
+        type="text"
+        required
+      >
+    </label>
+    <label for="position">
+      Position:
+      <input
+        name="position"
+        id="position"
+        type="text"
+        required
+      >
+    </label>
+    <label for="office">
+      Office:
+      <select
+        name="office"
+        id="office"
+        required
+      >
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+    <label for="age">
+      Age:
+      <input
+        name="age"
+        id="age"
+        type="number"
+        required
+      >
+    </label>
+    <label for="salary">
+      Salary:
+      <input
+        name="salary"
+        id="salary"
+        type="number"
+        required
+      >
+    </label>
+    <button type="submit">Save to table</button>
+  `;
+
+  return form;
+}
+
+function addRowAppending(form, table) {
```

Comment:
inside this function you 1) add functionality for row appending 2) validate the form with checks 3) add new employee to the table
you can create helper functions and transfer some code into them, it will look neater